schema/pgsql/schema.sql

```diff
@@ -242,111 +261,129 @@ CREATE TYPE severity AS ENUM ('ok', 'debug', 'info', 'notice', 'warning', 'err',
 CREATE TABLE event (
     id bigserial,
     time bigint NOT NULL,
-    object_id bytea NOT NULL REFERENCES object(id),
+    object_id bytea NOT NULL,
     type event_type NOT NULL,
     severity severity,
     message text,
     username citext,
     mute boolenum,
     mute_reason text,
 
-    CONSTRAINT pk_event PRIMARY KEY (id)
+    CONSTRAINT pk_event PRIMARY KEY (id),
+    CONSTRAINT fk_event_object FOREIGN KEY (object_id) REFERENCES object(id)
 );
 
 CREATE TABLE rule (
     id bigserial,
     name citext NOT NULL,
-    timeperiod_id bigint REFERENCES timeperiod(id),
+    timeperiod_id bigint,
     object_filter text,
     is_active boolenum NOT NULL DEFAULT 'y',
 
-    CONSTRAINT pk_rule PRIMARY KEY (id)
+    CONSTRAINT pk_rule PRIMARY KEY (id),
+    CONSTRAINT fk_rule_timeperiod FOREIGN KEY (timeperiod_id) REFERENCES timeperiod(id)
 );
 
 CREATE TABLE rule_escalation (
     id bigserial,
-    rule_id bigint NOT NULL REFERENCES rule(id),
+    rule_id bigint NOT NULL,
     position integer NOT NULL,
     condition text,
     name citext, -- if not set, recipients are used as a fallback for display purposes
-    fallback_for bigint REFERENCES rule_escalation(id),
+    fallback_for bigint,
 
     CONSTRAINT pk_rule_escalation PRIMARY KEY (id),
-
-    UNIQUE (rule_id, position),
-    CHECK (NOT (condition IS NOT NULL AND fallback_for IS NOT NULL))
+    CONSTRAINT key_rule_escalation_rule_position UNIQUE (rule_id, position),
+    CONSTRAINT onenull_rule_escalation_condition_fallback_for_check CHECK (NOT (condition IS NOT NULL AND fallback_for IS NOT NULL)),
```

Comment:
That will probably already be named in #191 due to a second `check` being added. Current name there is `ck_rule_escalation_fallback_has_no_fallback`.